### DISCUSSION TITLE
How to add variables that depends on other variabels?

### DISCUSSION THREAD

**Discussion by Rahimianes** [projectdiscovery/nuclei#932] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

Yeah, I have some domains that the usernames contain the first 10 characters of domains. 
I want to write the templates that all templates depend on the username of domains.
thanks

**Comment by ehsandeep** [comment, marked as answer]
Hi @Rahimianes,

Can you include more information for this, like sample input and expected template based on the input you wanted to make?

**Reply by Rahimianes** [reply]
Yeah Sure
I have many websites like the below domains:
domains.txt file contains domain1.com,domain2.com,domain3.com,.......,domain1000.com
the usernames for login into the hosting control panel like cPanel or DirectAdmin are the first 7 characters of domains. for example, see the below:

the username of domain1.com  is domain1
the username of domain2.com  is domain2
the username of domain3.com  is domain3
etc,....

the username of domain100.com  is domain10

now  I want to write the templates that all templates depend on the username of domains. 
for example, I need to write a template that has a variable for the username that username extracted from the 7 first characters of the domain.
Did you understand?
thanks

**Reply by ehsandeep** [reply]
@Rahimianes take this template as example - https://github.com/projectdiscovery/nuclei-templates/blob/master/fuzzing/header-command-injection.yaml
